### ISSUE TITLE
Incorrect serialization in production build

### ISSUE THREAD

**Issue by AndreyWV** [GillianPerard/typescript-json-serializer#145]
Sample code
```
@Serializable()
export class ApiPaginationFilters {

  public page: number = 1;

  @JsonProperty()
  public offset: number;

  constructor(
    @JsonProperty()
    public limit: number,
  ) {
  }

  public nextPage(): void {
    this.setPage(this.page + 1);
  }

  public setPage(page: number): void {
    this.page = page;
    this.offset = (this.page - 1) * this.limit;
  }

  public previousPage(): void {
    if (this.page === 1) {
      return;
    }
    this.setPage(this.page + 1);
  }

}

@Serializable()
export class EosagoArchiveParams extends ApiPaginationFilters {

  @JsonProperty()
  public state: string;

  @JsonProperty({
    name: 'start_date',
  })
  public startDate: string;

  @JsonProperty({
    name: 'end_date',
  })
  public endDate: string;
}

const params = new EosagoArchiveParams(5);
console.log(params);
console.log(serialize(params));

```

And this code has different console output in develop and production build of angular app.

Dev:
![dev-chrome](https://user-images.githubusercontent.com/29225873/124109454-d124b180-da6f-11eb-858e-541d443e28ba.png)

Prod:
![prod-chrome](https://user-images.githubusercontent.com/29225873/124109494-db46b000-da6f-11eb-85df-3e08f958d632.png)

Serialize instance of `ApiPaginationFilters` leads to same result

Declaring properties like this
```
@Serializable()
export class ApiPaginationFilters {

  public page: number = 1;

  @JsonProperty()
  public offset: number;

  @JsonProperty()
  public limit: number;

  constructor(
    limit: number,
  ) {
    this.limit = limit;
  }

  /* ... */

}

```
leads to same result too.

**Comment by GillianPerard** [comment]
Hi, sorry for the issue. I just tested it and yes there is a problem.

Let me check and fix asap.

**Comment by GillianPerard** [comment]
It seems it comes from Angular, I tested with React and it works properly.

I continue to investigate.

**Comment by AndreyWV** [comment]
I've create basic angular app with 100% reproducibility: https://github.com/AndreyWV/test-app-angular

**Comment by GillianPerard** [comment]
Yes I know, I meant that the problem happens only for Angular prod build.

**Comment by GillianPerard** [comment]
I published the version 3.4.2. 

Tell me if it works for you :)

**Comment by AndreyWV** [comment]
It works correctly now. Thanks! :smiley:
